src/navsat_transform.cpp

```diff
@@ -110,66 +118,68 @@ void NavSatTransform::run()
       datum_yaw = datum_vals[2];
     }
 
-    robot_localization::srv::set_datum::request request;
-    request.geo_pose.position.latitude = datum_lat;
-    request.geo_pose.position.longitude = datum_lon;
-    request.geo_pose.position.altitude = 0.0;
+    auto request = std::make_shared<robot_localization::srv::SetDatum::Request>();
+    request->geo_pose.position.latitude = datum_lat;
+    request->geo_pose.position.longitude = datum_lon;
+    request->geo_pose.position.altitude = 0.0;
     tf2::Quaternion quat;
     quat.setRPY(0.0, 0.0, datum_yaw);
-    request.geo_pose.orientation = tf2::toMsg(quat);
-    robot_localization::srv::set_datum::response response;
+    request->geo_pose.orientation = tf2::toMsg(quat);
+    auto response = std::make_shared<robot_localization::srv::SetDatum::Response>();
     datumCallback(request, response);
   }
 
-  auto odom_sub = node_->create_subscription<nav_msgs::msg::Odometry>(
-    "odometry/filtered", std::bind(&NavSatTransform::odomCallback, this, _1));
-  auto gps_sub = node_->create_subscription<sensor_msgs::msg::NavSatFix>(
-    "gps/fix", std::bind(&NavSatTransform::gpsFixCallback, this, _1));
+  odom_sub = this->create_subscription<nav_msgs::msg::Odometry>(
```

Comment:
The ROS1 version of this has the queue size set to `1`, I'd follow in line with that rather than `10` for this, gpd sub and IMU sub